hack/jenkins/save_build.sh

```diff
@@ -71,30 +70,52 @@ set -e
 
 sudo sed -i --regexp-extended "s/^(\s*)(appVersion\s*:\s*latest\s*$)/\1appVersion: $build_version/" $target_path/Chart.yaml
 
-#collecting source provenance data
+#collecting provenance data
 PRODUCT_NAME="Avi Kubernetes Operator"
 JENKINS_INSTANCE=$(echo $JENKINS_URL | sed -E 's/^\s*.*:\/\///g' | sed -E 's/:.*//g')
 COMP_UID="uid.obj.build.jenkins(instance='$JENKINS_INSTANCE',job_name='$JENKINS_JOB_NAME',build_number='$BUILD_NUMBER')"
-provenance_source_file="$WORKSPACE/provenance/source.json"
 
-# Function to run srp source provenance command
-function source_provenance() {
-    sudo /srp-tools/srp provenance source --scm-type git --name "$PRODUCT_NAME" --path ./ --saveto $provenance_source_file --comp-uid $COMP_UID --build-number ${version_build_num} --version $branch_version --all-ephemeral true --build-type release $@
-}
+# initialize credentials that are required for submission, Credentials value set by jenkins vault plugin
+sudo /srp-tools/srp config auth --client-id=${SRP_CLIENT_ID} --client-secret=${SRP_CLIENT_SECRECT}
+
+# initialize blank provenance in the working directory, $SRP_WORKING_DIR
+sudo /srp-tools/srp provenance init --working-dir $WORKSPACE/provenance
+sudo /srp-tools/srp provenance add-build jenkins --instance $JENKINS_INSTANCE --build-number $BUILD_NUMBER --job-name $JENKINS_JOB_NAME --working-dir $WORKSPACE/provenance
+
+# add an action for the golang build, importing the observations that were captured in the build-golang-app step
+sudo /srp-tools/srp provenance action start --name=ako-build --working-dir $WORKSPACE/provenance
+sudo /srp-tools/srp provenance action import-observation --name=ako-obs --file=$WORKSPACE/provenance/network_provenance.json --working-dir $WORKSPACE/provenance
+sudo /srp-tools/srp provenance action stop --working-dir $WORKSPACE/provenance
+
+# declare the git source tree for the build.  We refer to this declaration below when adding source inputs.
+sudo /srp-tools/srp provenance declare-source git --verbose --set-key=mainsrc --path=$WORKSPACE --branch=$BRANCH --working-dir $WORKSPACE/provenance
+
+CI_REGISTRY_IMAGE=10.79.172.11:5000/avi-buildops/ako
```

Comment:
Should we take `registry IP` as an input ? As this will be exposed to outside world, even though it is internal IP, we should take care of masking it?

Reply:
@akshayhavile I have addressed the review comment. The registries will be passed as args to the script and the same will have to be added the jenkins jobs.